internal/walletutil/store.go

```diff
@@ -65,10 +66,17 @@ func (s *EphemeralStore) SpendableSiacoinElements() []types.SiacoinElement {
 
 // Transactions returns all transactions relevant to the wallet, ordered
 // oldest-to-newest.
-func (s *EphemeralStore) Transactions() []wallet.Transaction {
+func (s *EphemeralStore) Transactions(since time.Time, max int) (txns []wallet.Transaction) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	return append([]wallet.Transaction(nil), s.txns...)
+	for _, txn := range s.txns {
+		if len(txns) == max {
+			return
+		} else if txn.Timestamp.Unix() >= since.Unix() {
```

Comment:
`txn.Timestamp.After(since)`

(technically, this is equivalent to `>`, not `>=`, but *also* technically, that might be the more correct interpretation of the word "since"...? Perhaps we should change "since" to "start" if we want it to unambiguously refer to a closed interval.)